### DISCUSSION TITLE
Issue in [@types/react] with useMemo

### DISCUSSION THREAD

**Discussion by JustFly1984** [DefinitelyTyped/DefinitelyTyped#67134] [Issues with a @types package]
Perfectly valid code
```
  const address = useMemo<ApiAddress | undefined>(() => {
    return addresses.find(
      ({ id }: ApiAddress): boolean => id === Number(selected)
    )
  }, [addresses, selected])
  ```
  After updating from 18.2.28 to 18.2.30 starts throwing an error "`ApiAddress | undefined` is not assignable to `ApiAddress`, undefined is not assignable to ApiAddress, and it happens in multiple files, where undefined expected as valid value, for example for style attribute.
  
  reverting @types/react@18.2.28 fixes an issue
  
  @johnnyreilly, @bbenezech, @pzavolinsky, @ericanderson, @DovydasNavickas, @theruther4d, @guilhermehubner, @ferdaber, @jrakotoharisoa, @pascaloliv, @Hotell, @franklixuefei, @Jessidhia, @saranshkataria, @lukyth, @eps1lon, @zieka, @dancerphil, @dimitropoulos, @disjukr, @vhfmag, @hellatan, @priyanshurav, @Semigradsky.

[a comment by an automated account is not shown]

**Comment by dancerphil** [comment]
See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67076

Revert progress: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67135

Further discussion: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67132
